File: frontend/src/types/Task.ts
export enum TaskStatus {
  PENDING = 'PENDING',
  COMPLETED = 'COMPLETED'
}

export interface Task {
  id: number;
  title: string;
  description?: string;
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTaskInput {
  title: string;
  description?: string;
}

export interface UpdateTaskStatusInput {
  id: number;
  status: TaskStatus;
}

export interface TaskPayload {
  task?: Task;
  errorMessage?: string;
  success: boolean;
}

File: backend/TodoApp.Core/Entities/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TodoApp.Core.Entities;

public class Task
{
    public int Id { get; set; }
    
    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;
    
    [MaxLength(1000)]
    public string? Description { get; set; }
    
    public TaskStatus Status { get; set; } = TaskStatus.Pending;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum TaskStatus
{
    Pending = 0,
    Completed = 1
}